spec/unit/provider/package/cab_spec.rb

```diff
@@ -0,0 +1,191 @@
+#
+# Author:: Vasundhara Jagdale (<vasundhara.jagdale@msystechnologies.com>)
+# Copyright:: Copyright 2008-2016, Chef Software, Inc.
+# License:: Apache License, Version 2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+require "spec_helper"
+
+describe Chef::Provider::Package::Cab do
+  let(:timeout) {}
+
+  let(:new_resource) { Chef::Resource::CabPackage.new("windows_test_pkg") }
+
+  let(:provider) do
+    node = Chef::Node.new
+    events = Chef::EventDispatch::Dispatcher.new
+    run_context = Chef::RunContext.new(node, {}, events)
+    Chef::Provider::Package::Cab.new(new_resource, run_context)
+  end
+
+  let(:installed_package_list_stdout) do
+    <<-EOF
+Packages listing:
+Package Identity : Package_for_KB2999486~31bf3856ad364e35~amd64~~6.1.9768.0
+Package Identity : Package_for_KB2994825~31bf3856ad364e35~amd64~~6.1.7601.0
+    EOF
+  end
+
+  let(:package_version_stdout) do
+    <<-EOF
+Package information:
+Package Identity : Package_for_KB2664825~31bf3856ad364e35~amd64~~6.1.3.0
+State : Installed
+Dependency : Language Pack
+The operation completed successfully
+    EOF
+  end
+
+  before do
+    new_resource.source = "C:\\Temp\\Test6.1-KB2664825-v3-x64.cab"
+    installed_package_list_obj = double(stdout: installed_package_list_stdout)
+    allow(provider).to receive(:dism_command).with("/Get-Packages").and_return(installed_package_list_obj)
+    package_version_obj = double(stdout: package_version_stdout)
+    allow(provider).to receive(:dism_command).with("/Get-PackageInfo /PackagePath:\"#{new_resource.source}\"").and_return(package_version_obj)
+  end
+
+  def allow_package_info(package_path = nil, package_name = nil)
+    get_pakage_info_stdout = <<-EOF
```

Comment:
there are a few misspellings in variable names here, so I'm not sure if they were intentional. But we should spell them correctly so the next person to work on the code doesn't have to remember how to correctly misspell them.

*get_package_info_stdout